FlowCal/plot.py

```diff
@@ -1058,21 +1141,33 @@ def scatter2d(data_list,
         plt.ylabel(data_plot.channels[1])
 
     # Set scale of axes
-    plt.gca().set_xscale(xscale)
-    plt.gca().set_yscale(yscale)
+    plt.gca().set_xscale(xscale, data=data_list, channel=channels[0])
+    plt.gca().set_yscale(yscale, data=data_list, channel=channels[1])
 
-    # Set plot limits if specified, else extract range from data_plot
+    # Set plot limits if specified, else extract range from data_list
     # Use data_plot.hist_bins with one single bin
-    if xlim is not None:
-        plt.xlim(xlim)
-    elif hasattr(data_plot, 'hist_bins') and \
-            hasattr(data_plot.hist_bins, '__call__'):
-        plt.xlim(data_plot.hist_bins(channels=0, nbins=1, scale=xscale))
-    if ylim is not None:
-        plt.ylim(ylim)
-    elif hasattr(data_plot, 'hist_bins') and \
-            hasattr(data_plot.hist_bins, '__call__'):
-        plt.ylim(data_plot.hist_bins(channels=1, nbins=1, scale=yscale))
+    if xlim is None:
+        xlim = [np.inf, -np.inf]
+        for data in data_list:
+            if hasattr(data, 'hist_bins') and \
+                    hasattr(data.hist_bins, '__call__'):
+                xlim_data = data.hist_bins(channels=channels[0],
+                                           nbins=1,
+                                           scale=xscale)
```

Comment:
`hist_bins` automatically handles the case in which `log` scaling is requested, but the lower limit of `range` is zero (you can't log zero).

Another way to handle this would be to get the range, explicitly check if the lower value is zero, and take appropriate action. But this happened frequently enough that I though using `hist_bins`  in all these instances was a better option.